### PR TITLE
Update Deep Learning Examples automation to handle containerd

### PR DESCRIPTION
**Pull request by ajdecon**
## Summary

Our automation for Deep Learning Examples depended on using the host's Docker daemon to build the example container. Now that we are using containerd, there is no Docker daemon!

This PR modifies our automation to use [kaniko](https://github.com/GoogleContainerTools/kaniko) for the image build step, instead of the local Docker daemon.

This does result in adding a little additional complication. The kaniko pod can only see the registry on its local ClusterIP service, but containerd can only see the registry using the NodePort. So we have to address these differently:

```
registryName: registry.local
registryPort: 31500
internalRegistryName: docker-registry.deepops-docker
internalRegistryPort: 5000
```

Addresses #1144 

## TODO

- [x] Update to use kaniko
- [x] Update documentation
- [x] Add Jenkins tests

## Test plan

Run a deployment of a Deep Learning Example:

```
+ helm install --wait --timeout 1200s pytorch-detection-ssd workloads/examples/k8s/deep-learning-examples --set exampleName=pytorch-detection-ssd
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /home/ubuntu/.kube/config
WARNING: Kubernetes configuration file is world-readable. This is insecure. Location: /home/ubuntu/.kube/config
NAME: pytorch-detection-ssd
LAST DEPLOYED: Wed Mar 30 15:19:59 2022
NAMESPACE: default
STATUS: deployed
REVISION: 1
NOTES:
--------------------------------------------------------------------------------
Installation Complete!
    App:             pytorch-detection-ssd
    registryName:    registry.local
    registryPort:    31500
    jupyterNodePort: 30888
    resources:
            limits:
              nvidia.com/gpu: 1

1. Monitor build progress of the InitContainer:
    kubectl logs pod/pytorch-detection-ssd-0 -c pytorch-detection-ssd-init -f

2. Monitor the complete application:
    kubectl get all -l app=pytorch-detection-ssd -o wide

3. Capture the hostname of the JupyterLab Service on port 30888
    kubectl get pod/pytorch-detection-ssd-0 -o jsonpath='{.spec.nodeName}'
--------------------------------------------------------------------------------
```

Added Jenkins tests.

### CODE REVIEW

**Comment by ajdecon** [comment]
Jenkins test passing

image

File: workloads/jenkins/Jenkinsfile [hunk 1]
```diff
@@ -83,6 +83,11 @@ pipeline {
             timeout 180 bash -x ./workloads/jenkins/scripts/test-dashboard.sh
           '''
 
+          echo "Test running a Deep Learning Example"
```
Comment (supertetelman) [inline workloads/jenkins/Jenkinsfile:86]: How long does this test take? I've been trying to keep the PR Jenkinsfile light based on previous feedback.

File: workloads/jenkins/scripts/test-dle-deployment.sh [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	"${JENKINS_DLE}" \
+	workloads/examples/k8s/deep-learning-examples \
+	--set exampleName="${JENKINS_DLE}"
+
+if [ $? -ne 0 ]; then
+	echo "Failed to deploy DLE"
+	exit 1
+fi
+
+sleep 60
```
Comment (supertetelman) [inline workloads/jenkins/scripts/test-dle-deployment.sh:30]: It'd probably be better to copy/paste the polling we do in the other tests, but 60 seconds is probably fine if we are only running this once.

**Review by supertetelman — DISMISSED** [review]
LGTM, some optimizations we could make in the tests, but otherwise looks good.

File: workloads/jenkins/Jenkinsfile (hunk 1 above)
Comment (ajdecon) [reply workloads/jenkins/Jenkinsfile]: Good call on removing this -- it adds 11-12 minutes.

**Pull request merged**